paddle/fluid/framework/op_proto_maker.cc

```diff
@@ -75,6 +75,10 @@ void OpProtoAndCheckerMaker::operator()(proto::OpProto* proto,
            static_cast<int>(OpRole::kOptimize) |
                static_cast<int>(OpRole::kLRSched),
            static_cast<int>(OpRole::kCollective),
+           static_cast<int>(OpRole::kCollective) |
+               static_cast<int>(OpRole::kBackward),
+           static_cast<int>(OpRole::kCollective) |
+               static_cast<int>(OpRole::kOptimize),
```

Comment:
Op role will increase the complexity of the Graph and Program analysis. I do not recommend adding new o prole.

Reply:
i agree with u, and i will try to remove newly added op roles. However, as collective ops swap data among trainers, their behaviors is different from backward and optimize ops more or less, i will create a new pr to discuss this topic if necessary. 3ks.

Reply:
done